src/components/PreJoinScreens/DeviceSelectionScreen/DeviceSelectionScreen.tsx

```diff
@@ -71,6 +72,21 @@ export default function DeviceSelectionScreen({ name, roomName, setStep }: Devic
     });
   };
 
+  if (isFetching || isConnecting) {
+    return (
+      <Grid container justify="center" spacing={2} alignItems="center" direction="column" style={{ height: '100%' }}>
+        <Grid item>
+          <CircularProgress variant="indeterminate" />
+        </Grid>
+        <Grid item>
+          <Typography variant="body2" style={{ fontWeight: 'bold', fontSize: '16px' }}>
+            Joining Meeting
+          </Typography>
+        </Grid>
```

Comment:
Nit - I think these work the same as plain divs:

```suggestion
        <div>
          <CircularProgress variant="indeterminate" />
        </div>
        <div>
          <Typography variant="body2" style={{ fontWeight: 'bold', fontSize: '16px' }}>
            Joining Meeting
          </Typography>
        </div>
```